metabase/metabase
# Add Read-Only Notebook View

Resolves https://github.com/metabase/metabase/issues/29456
Resolves https://github.com/metabase/metabase/issues/28531
Resolves https://github.com/metabase/metabase/issues/28639

### Description

Until now, the audit log was showing the old/deprecated GUI query builder (instead of the notebook) to graphically represent queries not built with SQL.  Most importantly, this component started throwing page-crashing errors in the audit log. Instead of fixing deprecated code, I opted to replace that view with a simple read-only version of the notebook query builder.

1. Creates a New `QueryViewer` container that pulls in appropriate metadata for a query and shows either the notebook editor or native query editor in read only mode, as appropriate.
2. Creates a new `ReadOnlyNotebook` component which passes down `readOnly` props through the notebook editor to hide various UI elements that suggest you should be clicking on them.
3. Adds unit test coverage, which I think is the first and only unit test coverage for the notebook editor 🥳 

**Before**
(when it didn't crash)
![Screen Shot 2023-03-24 at 11 02 56 AM](https://user-images.githubusercontent.com/30528226/227596392-9a06b87e-d8f3-4490-ab12-2dd95c26aa4c.png)


**After**

![Screen Shot 2023-03-23 at 3 23 28 PM](https://user-images.githubusercontent.com/30528226/227585741-87b59806-b87a-446d-9be9-c90db350b6b5.png)

### How to verify

1. Startup metabase enterprise
2. open a new question and create a GUI question with filters, aggregations, limits, etc.
3. open another new question and create a sql question
5. Open the audit log by going to admin settings > audit > audit log
6. click on the two ad-hoc queries you made and see that both notebook and native queries display properly
7. ensure that both query viewers are non-interactive
8. refresh the page on the audit page and ensure that the query displays still display properly there

### Checklist



- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/metabase/metabase/29525)
